File: fe-smart-job/src/services/chatService.js
import axiosClient from '../config/axiosClient';

export const chatService = {
  getOrCreateConversation: (data) => axiosClient.post('/chat/conversations/initiate', data),

  sendMessage: (data) => axiosClient.post('/chat/messages', data),
  
  getMyConversations: () => axiosClient.get('/chat/conversations'),
  
  getMessagesByConversationId: (conversationId) => 
    axiosClient.get(`/chat/conversations/${conversationId}/messages`),
    
  markConversationAsRead: (conversationId) => 
    axiosClient.put(`/chat/conversations/${conversationId}/read`),
};

File: be-smart-job/src/main/java/be_smart_job/service/social/interfaces/ChatService.java
package be_smart_job.service.social.interfaces;

import be_smart_job.dto.req.social.GetOrCreateConversationRequest;
import be_smart_job.dto.req.social.SendMessageRequest;
import be_smart_job.dto.res.social.ConversationResponse;
import be_smart_job.dto.res.social.MessageResponse;

import java.util.List;

public interface ChatService {
    ConversationResponse getOrCreateConversation(GetOrCreateConversationRequest request);
    MessageResponse sendMessage(SendMessageRequest request);
    List<ConversationResponse> getMyConversations();
    List<MessageResponse> getMessagesByConversationId(String conversationId);
    void markConversationAsRead(String conversationId);
}